app/src/main/java/com/github/fribourgsdp/radio/auth/GoogleSignIn.kt

```diff
@@ -0,0 +1,122 @@
+import com.github.fribourgsdp.radio.data.view.UserProfileActivity
+import com.github.fribourgsdp.radio.external.google.GoogleSignInActivity
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import java.lang.Exception
+
+
+
+class GoogleSignIn(val activity: UserProfileActivity, serverClientId : String) {
+
+    private val googleSignInOptions : GoogleSignInOptions
+    private val googleSignInClient : GoogleSignInClient
+    private val launcher : ActivityResultLauncher<Intent>
+    private lateinit var firebaseAuth : FirebaseAuth
+
+    private companion object {
+        private const val TAG = "GOOGLE_SIGN_TAG"
+    }
+
+    init{
+        googleSignInOptions = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(serverClientId)
+            .requestEmail()
+            .build()
+        googleSignInClient = GoogleSignIn.getClient(activity, googleSignInOptions)
+
+
+        launcher =
+            activity.registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+                //handle intent result here
+                if (result.resultCode == Activity.RESULT_OK) {
+                    val accountTask = GoogleSignIn.getSignedInAccountFromIntent(result.data)
+                    try {
+                        //Google SignIn success, now authenticate with firebase
+                        val account = accountTask.getResult(ApiException::class.java)
+                        val credential = GoogleAuthProvider.getCredential(account.idToken, null)
+                        this.firebaseAuthWithCredentitial(credential, firebaseAuth)
+                    } catch (e: Exception) {
+                        //failed Google SignIn
+                        Log.d(TAG, "onActivityResult: ${e.message}")
+                    }
+                } else {
+                    //cancelled
+                    Toast.makeText(activity, "Cancelled", Toast.LENGTH_SHORT).show()
+                }
+            }
+    }
+
+    fun signIn(firebaseAuth : FirebaseAuth = FirebaseAuth.getInstance()){
+        //init firebase auth
+        this.firebaseAuth = firebaseAuth
+
+        if(checkUser()) {
+            activity.loginFromGoogle(GoogleSignInResult.ALREADY_LOGIN)
+        }else{
+            val intent = googleSignInClient.signInIntent
+            launcher.launch(intent)
+        }
+    }
+
+    fun firebaseAuthWithCredentitial(credential: AuthCredential, firebaseAuth: FirebaseAuth) {
+        firebaseAuth.signInWithCredential(credential).addOnSuccessListener { authResult ->
+            //login success
+            var isNew = 0
```

Comment:
What is the purpose of this var? It doesn't seem to be used.

Reply:
yes you are right